ansible/roles/code/tasks/main.yml

```diff
@@ -20,11 +37,26 @@
     dest: /usr/local/code/faasm
     update: yes
 
-- name: "Directory ownership"
+# Not supported by ansible: recursively change ownership for dirs and files
+- name: "Set directory ownership to faasm"
   become: yes
-  file:
-    path: "/usr/local/code/faasm"
-    state: directory
-    owner: "{{ ansible_env.USER }}"
-    group: "{{ ansible_env.USER }}"
-    mode: 0755
\ No newline at end of file
+  command: |
+    chown -c -R :faasm "{{ item }}"
+  register: chown_status
+  changed_when: chown_status.stdout != ""
+  with_items:
+    - /usr/local/code/faasm
+  args:
+    warn: no
```

Comment:
Wow, this is grim, but understandable. Is there some gitconfig somewhere to ignore permission changes?

Reply:
I tried this exact thing, unfortunately changing the git config doesn't work for submodules, even when made `--global` or `--system`. Plus it is heavily recommended not to to touch this configuration, especially not to set it user-wide or system-wide.

The best solution in my mind was to create a `faasm` user on which we do everything. This is me trying not to have to go down to such extend because that could break some of the stuff you have so far and can become annoying for containers and stuff, but I think we might have to at some point.

Let's say we try this out and if it doesn't work I'll fix it, as this seems to work fine so far. I'm scared python, `/usr/local/faasm` and everything not on `/usr/local/code/faasm` might be an issue though at some point.